base/array.jl

```diff
@@ -1210,11 +1188,13 @@ julia> findprev(isodd, A, 3)
 2
 ```
 """
-function findprev(testf::Function, A, start::Integer)
-    for i = start:-1:1
-        testf(A[i]) && return i
+function findprev(testf::Function, A, idx::Integer)
+    while idx > 0
+        elem, _ = next(A, idx)
+        testf(elem) && return idx
+        idx = prevind(A, idx)
```

Comment:
This seems to assume that all iterables implement `prevind`, when you have only defined it for `AbstractArray` (and strings).